src/admin/datatypes/composite-datatypes/index.md

```diff
@@ -35,44 +35,44 @@ class BasicComposite(Text):
-
-There are cases when it is desireable for the composite filenames to have varying names, but be of a similar form; for an example of this see Rgenetics below.
-
-Before continuing with this section, be sure to read and understand [AddingDatatypes](Admin%2FDatatypes%2FAdding+Datatypes).
-
-Composite datatypes can be used as a more structured way to contain individual history items which are composed of multiple files. The Rgenetics package for Galaxy has been implemented using Composite Datatypes; for real-life examples, examine the configuration files (particularly lib/galaxy/datatypes/genetics.py) included with the distribution.
-
-In composite datatypes, there is one "primary" data file and any number of specified "component" files. The primary data file is a base dataset object and the component files are all located in a directory associated with that base dataset. There are two types of composite datatypes: **basic** and **auto\_primary\_file**. In basic composite datatypes, the primary data file must be written by tools or provided directly during upload. In auto\_primary\_file composite datatypes, the primary data file is generated automaticaly by the Framework; all tool-writable or uploadable files are stored in the directory associated with the primary file.
-
-### Creating Composite Datatypes
-
-A datatype can be set to composite by setting the <tt>composite_type</tt> flag. There are 3 valid options:
-
-- None (not a composite datatype) 
-- 'basic' 
-- 'auto\_primary\_file' 
-
-### Defining Auto Primary File Composite Datatypes
-
-The example below defines an auto primary file composite datatype which is composed of 2 component files along with a framework generated file. In this example the primary file is an html page containing download links to the individual result files. The two component files (results.txt, results.dat) contain the results; results.txt is a text file and is handled as such during upload whereas results.dat is flaged as binary, allowing a binary file to be uploaded for that component. During upload, 2 sets of file upload/paste interfaces are provided to the user. A file must be provided for results.txt whereas results.dat is flaged as optional so may be left blank.
-
-In this composite type, the primary file is generated using the <tt>generate_primary_file</tt> method specified in the datatype definition. The <tt>generate_primary_file</tt> method here loops through the defined components and creates a link to each.
-
-```
+${os.path.join( input1.extra_files_path, 'results.dat' )} $output1</command>
+```
+
+If a tool is aware of the file names for a datatype, then only input1.extra\_files\_path needs to be provided.
+
+There are cases when it is desireable for the composite filenames to have varying names, but be of a similar form; for an example of this see Rgenetics below.
+
+Before continuing with this section, be sure to read and understand [AddingDatatypes](Admin%2FDatatypes%2FAdding+Datatypes).

```

Comment:
This and any of the other links with %2F/etc in them probably need to be updated.